From c127b6d1f30234707ebe7e444ab2f2d91b31953c Mon Sep 17 00:00:00 2001
From: Vishesh Handa <me@vhanda.in>
Date: Thu, 27 Mar 2014 19:31:40 +0100
Subject: [PATCH 05/56] Commit Queue: Stop both the timers after committing

---
 src/file/commitqueue.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/file/commitqueue.cpp b/src/file/commitqueue.cpp
index fafaaa1..b59bb5d 100644
--- a/src/file/commitqueue.cpp
+++ b/src/file/commitqueue.cpp
@@ -86,4 +86,7 @@ void Baloo::CommitQueue::commit()
     kDebug() << "SQL Committed";
 
     m_db->xapianDatabase()->commit();
+
+    m_smallTimer.stop();
+    m_largeTimer.stop();
 }
-- 
1.9.1

