From 2d4ef34670a57a5076511cf6f1bea16cff73ffad Mon Sep 17 00:00:00 2001
From: Vishesh Handa <me@vhanda.in>
Date: Thu, 27 Mar 2014 19:32:09 +0100
Subject: [PATCH 06/48] BasicIndexingQueueTest: Print io usage by the basic
 indexing

---
 src/file/tests/basicindexingqueuetest.cpp | 22 ++++++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/src/file/tests/basicindexingqueuetest.cpp b/src/file/tests/basicindexingqueuetest.cpp
index bb28ca9..60fbc30 100644
--- a/src/file/tests/basicindexingqueuetest.cpp
+++ b/src/file/tests/basicindexingqueuetest.cpp
@@ -55,6 +55,28 @@ int main(int argc, char** argv)
     timer.start();
     int ret = app.exec();
 
+    commitQueue.commit();
     qDebug() << "Elapsed:" << timer.elapsed();
+
+    // Print the io usage
+    QFile file("/proc/self/io");
+    file.open(QIODevice::ReadOnly | QIODevice::Text);
+
+    QTextStream fs(&file);
+    QString str = fs.readAll();
+
+    QTextStream stream(&str);
+    while (!stream.atEnd()) {
+        QString str = stream.readLine();
+        if (str.startsWith("rchar")) {
+            ulong amt = str.mid(QString("rchar: ").size()).toULong();
+            qDebug() << "Read:" << amt / 1024  << "kb";
+        }
+        if (str.startsWith("wchar")) {
+            ulong amt = str.mid(QString("wchar: ").size()).toULong();
+            qDebug() << "Written:" << amt / 1024  << "kb";
+        }
+    }
+
     return ret;
 }
-- 
1.9.1

