From 29a334fe6dfb46fb36d44abd7adae06b47cba723 Mon Sep 17 00:00:00 2001
From: Vishesh Handa <me@vhanda.in>
Date: Sat, 29 Mar 2014 13:02:09 +0100
Subject: [PATCH 14/56] Use KLocalizedString instead of ki18n("")

---
 src/file/extractor/main.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/file/extractor/main.cpp b/src/file/extractor/main.cpp
index ec9dce0..4be2c24 100644
--- a/src/file/extractor/main.cpp
+++ b/src/file/extractor/main.cpp
@@ -51,7 +51,7 @@ int main(int argc, char* argv[])
     options.add("debug", ki18n("Print the data being indexed"));
     options.add("bdata", ki18n("Print the QVariantMap in Base64 encoding"));
     // FIXME: Set a proper string after the freeze. This option is just for debugging
-    options.add("db <url>", ki18n(""));
+    options.add("db <url>", KLocalizedString());
 
     KCmdLineArgs::addCmdLineOptions(options);
     const KCmdLineArgs* args = KCmdLineArgs::parsedArgs();
-- 
1.9.1

