From 36ddb43f868a16688d72b201bd09642b0f988f80 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Dan=20Vr=C3=A1til?= <dvratil@redhat.com>
Date: Sun, 30 Mar 2014 11:50:52 +0200
Subject: [PATCH 16/48] Catch DatabaseError exception in XapianSearchStore

---
 src/xapian/xapiansearchstore.cpp | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/xapian/xapiansearchstore.cpp b/src/xapian/xapiansearchstore.cpp
index ff82a6e..59c035b 100644
--- a/src/xapian/xapiansearchstore.cpp
+++ b/src/xapian/xapiansearchstore.cpp
@@ -199,7 +199,12 @@ int XapianSearchStore::exec(const Query& query)
         return 0;
 
     QMutexLocker lock(&m_mutex);
-    m_db->reopen();
+    try {
+        m_db->reopen();
+    } catch (Xapian::DatabaseError& e) {
+        kWarning() << "Failed to reopen database" << dbPath() << ":" <<  QString::fromStdString(e.get_msg());
+        return 0;
+    }
 
     QTime queryGenerationTimer;
     queryGenerationTimer.start();
-- 
1.9.1

