From e54afe1e2d131d46a2ac6df4303aadc41507c0c2 Mon Sep 17 00:00:00 2001
From: Vishesh Handa <me@vhanda.in>
Date: Sat, 29 Mar 2014 15:31:49 +0100
Subject: [PATCH 18/48] KCM: Use dbus asyncCall instead of send

---
 src/file/kcm/kcm.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/file/kcm/kcm.cpp b/src/file/kcm/kcm.cpp
index 685b6e1..f40d365 100644
--- a/src/file/kcm/kcm.cpp
+++ b/src/file/kcm/kcm.cpp
@@ -34,6 +34,7 @@
 #include <QProcess>
 #include <QDBusConnection>
 #include <QDBusMessage>
+#include <QDBusPendingCall>
 
 K_PLUGIN_FACTORY(BalooConfigModuleFactory, registerPlugin<Baloo::ServerConfigModule>();)
 K_EXPORT_PLUGIN(BalooConfigModuleFactory("kcm_baloofile", "kcm_baloofile"))
@@ -119,7 +120,7 @@ void ServerConfigModule::save()
                                                               QLatin1String("org.kde.baloo.file"),
                                                               QLatin1String("quit"));
 
-        QDBusConnection::sessionBus().send(message);
+        QDBusConnection::sessionBus().asyncCall(message);
     }
 
     // Start cleaner
-- 
1.9.1

