From 6493efc09e69bfd974b86937017eaa9210cf7420 Mon Sep 17 00:00:00 2001
From: Vishesh Handa <me@vhanda.in>
Date: Sat, 29 Mar 2014 17:45:27 +0100
Subject: [PATCH 21/48] Cleaner: Commit the changes every 1000 documents

---
 src/file/cleaner/cleaner.cpp | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/file/cleaner/cleaner.cpp b/src/file/cleaner/cleaner.cpp
index 1cfec6f..d64a39d 100644
--- a/src/file/cleaner/cleaner.cpp
+++ b/src/file/cleaner/cleaner.cpp
@@ -50,6 +50,7 @@ void Cleaner::start()
 
     FileIndexerConfig config;
 
+    int numDocuments = 0;
     while (query.next()) {
         int id = query.value(0).toInt();
         QString url = query.value(1).toString();
@@ -75,6 +76,12 @@ void Cleaner::start()
             q.addBindValue(id);
             q.exec();
             m_commitQueue->remove(id);
+
+            numDocuments++;
+        }
+
+        if (numDocuments >= 1000) {
+            m_commitQueue->commit();
         }
     }
 
-- 
1.9.1

