From 9d1c2d8c8e7d195b7d0648af0ca9bf6c874780fa Mon Sep 17 00:00:00 2001
From: Vishesh Handa <me@vhanda.in>
Date: Tue, 1 Apr 2014 11:34:40 +0200
Subject: [PATCH 25/48] IndexerConfigTest: Show both fast and slow mimetype

---
 src/file/tests/indexerconfigtest.cpp | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/src/file/tests/indexerconfigtest.cpp b/src/file/tests/indexerconfigtest.cpp
index acf9041..5b487b8 100644
--- a/src/file/tests/indexerconfigtest.cpp
+++ b/src/file/tests/indexerconfigtest.cpp
@@ -66,11 +66,14 @@ int main(int argc, char** argv)
     bool shouldIndex = config.shouldBeIndexed(url.toLocalFile());
 
     QString mimetype = KMimeType::findByUrl(url)->name();
-    bool shouldIndexMimetype = config.shouldMimeTypeBeIndexed(mimetype);
+    QString fastMimetype = KMimeType::findByUrl(url, 0, true, true)->name();
+
+    bool shouldIndexMimetype = config.shouldMimeTypeBeIndexed(fastMimetype);
     std::cout << url.toLocalFile().toUtf8().constData() << "\n"
               << "Should Index: " << std::boolalpha << shouldIndex << "\n"
               << "Should Index Mimetype: " << std::boolalpha << shouldIndexMimetype << "\n"
-              << "Mimetype: " << mimetype.toUtf8().constData() << std::endl;
+              << "Fast Mimetype: " << fastMimetype.toUtf8().constData() << std::endl
+              << "Slow Mimetype: " << mimetype.toUtf8().constData() << std::endl;
 
     return 0; //app.exec();
 }
-- 
1.9.1

