From 562f1706f1b6b548fd7d74fec76f47012aeb4b91 Mon Sep 17 00:00:00 2001
From: Vishesh Handa <me@vhanda.in>
Date: Wed, 2 Apr 2014 12:13:15 +0200
Subject: [PATCH 35/56] XapianDatabase: Add missing slots declaration

---
 src/xapian/xapiandatabase.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/xapian/xapiandatabase.h b/src/xapian/xapiandatabase.h
index 24d489d..d982761 100644
--- a/src/xapian/xapiandatabase.h
+++ b/src/xapian/xapiandatabase.h
@@ -39,6 +39,7 @@ class BALOO_XAPIAN_EXPORT XapianDatabase : public QObject
 public:
     XapianDatabase(const QString& path);
 
+public Q_SLOTS:
     void replaceDocument(uint id, const Xapian::Document& doc);
     void deleteDocument(uint id);
 
@@ -49,6 +50,7 @@ public:
      */
     void commit();
 
+public:
     XapianDocument document(uint id);
 
     /**
-- 
1.9.1

