From 3fe4d2e97aabbb878b30a1d572cebb5d65fc2848 Mon Sep 17 00:00:00 2001
From: Vishesh Handa <me@vhanda.in>
Date: Wed, 2 Apr 2014 12:13:32 +0200
Subject: [PATCH 36/56] IndexSchduler: Decrease fileIQ delay when idle

---
 src/file/indexscheduler.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/file/indexscheduler.cpp b/src/file/indexscheduler.cpp
index ce4294f..6e2d354 100644
--- a/src/file/indexscheduler.cpp
+++ b/src/file/indexscheduler.cpp
@@ -266,8 +266,10 @@ bool IndexScheduler::scheduleFileQueue()
             return false;
         case State_UserIdle:
             m_fileIQ->setDelay(0);
+            return true;
         case State_Normal:
             m_fileIQ->setDelay(500);
+            return true;
         default:
             return true;
     }
-- 
1.9.1

