From 707f61f3c4134b64bdd923cf4dc5199219b79c01 Mon Sep 17 00:00:00 2001
From: Vishesh Handa <me@vhanda.in>
Date: Mon, 7 Apr 2014 18:20:21 +0200
Subject: [PATCH 49/56] XapianDatabase: Catch more exceptions

BUG: 332403
---
 src/xapian/xapiandatabase.cpp | 16 ++++++++++++----
 1 file changed, 12 insertions(+), 4 deletions(-)

diff --git a/src/xapian/xapiandatabase.cpp b/src/xapian/xapiandatabase.cpp
index a509731..2fda1de 100644
--- a/src/xapian/xapiandatabase.cpp
+++ b/src/xapian/xapiandatabase.cpp
@@ -62,7 +62,11 @@ void XapianDatabase::replaceDocument(uint id, const XapianDocument& doc)
 void XapianDatabase::replaceDocument(uint id, const Xapian::Document& doc)
 {
     if (m_writeOnly) {
-        m_wDb.replace_document(id, doc);
+        try {
+            m_wDb.replace_document(id, doc);
+        }
+        catch (const Xapian::Error&) {
+        }
         return;
     }
     m_docsToAdd << qMakePair(id, doc);
@@ -74,7 +78,7 @@ void XapianDatabase::deleteDocument(uint id)
         try {
             m_wDb.delete_document(id);
         }
-        catch (const Xapian::DocNotFoundError&) {
+        catch (const Xapian::Error&) {
         }
         return;
     }
@@ -96,7 +100,11 @@ void XapianDatabase::commit()
 
     kDebug() << "Adding:" << m_docsToAdd.size() << "docs";
     Q_FOREACH (const DocIdPair& doc, m_docsToAdd) {
-        wdb.replace_document(doc.first, doc.second);
+        try {
+            wdb.replace_document(doc.first, doc.second);
+        }
+        catch (const Xapian::Error&) {
+        }
     }
 
     kDebug() << "Removing:" << m_docsToRemove.size() << "docs";
@@ -104,7 +112,7 @@ void XapianDatabase::commit()
         try {
             wdb.delete_document(id);
         }
-        catch (const Xapian::DocNotFoundError&) {
+        catch (const Xapian::Error&) {
         }
     }
 
-- 
1.9.1

