From 76bc0308863ac5cdbc82ec7758e90fbbc014a810 Mon Sep 17 00:00:00 2001
From: Montel Laurent <montel@kde.org>
Date: Tue, 8 Apr 2014 07:25:04 +0200
Subject: [PATCH 50/56] Prepare 4.13.0

---
 CMakeLists.txt | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 11f863e..1e5b278 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,8 +1,8 @@
 project(baloo)
 
 set(BALOO_VERSION_MAJOR 4)
-set(BALOO_VERSION_MINOR 12)
-set(BALOO_VERSION_RELEASE 97)
+set(BALOO_VERSION_MINOR 13)
+set(BALOO_VERSION_RELEASE 0)
 set(BALOO_VERSION "${BALOO_VERSION_MAJOR}.${BALOO_VERSION_MINOR}.${BALOO_VERSION_RELEASE}")
 
 
@@ -11,7 +11,7 @@ set(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake/modules" ${CMAKE_MODULE_PATH})
 find_package(KDE4 REQUIRED)
 include(KDE4Defaults)
 
-find_package(KdepimLibs 4.12.97 CONFIG)
+find_package(KdepimLibs 4.13.0 CONFIG)
 set_package_properties(KdepimLibs PROPERTIES
     DESCRIPTION "The KDEPIM libraries"
     URL "http://www.kde.org"
-- 
1.9.1

