From 798fbc52bbc5f5b8d1635eac90340e6f3aa6c374 Mon Sep 17 00:00:00 2001
From: Vishesh Handa <me@vhanda.in>
Date: Tue, 8 Apr 2014 15:23:14 +0200
Subject: [PATCH 54/56] Email Test: Print io debug information

---
 src/pim/agent/tests/emailtest.cpp | 40 +++++++++++++++++++++++++++++++++++++++
 1 file changed, 40 insertions(+)

diff --git a/src/pim/agent/tests/emailtest.cpp b/src/pim/agent/tests/emailtest.cpp
index 5746975..9166268 100644
--- a/src/pim/agent/tests/emailtest.cpp
+++ b/src/pim/agent/tests/emailtest.cpp
@@ -24,6 +24,7 @@
 
 #include <QApplication>
 #include <QTimer>
+#include <QFile>
 #include <KDebug>
 
 #include <Akonadi/CollectionFetchJob>
@@ -158,6 +159,45 @@ void App::slotIndexed()
     kDebug() << "Emails:" << m_numEmails;
     kDebug() << "Total Time:" << m_totalTime.elapsed()/1000.0 << " seconds";
     kDebug() << "Index Time:" << m_indexTime/1000.0 << " seconds";
+
+    // Print the io usage
+    QFile file("/proc/self/io");
+    file.open(QIODevice::ReadOnly | QIODevice::Text);
+
+    QTextStream fs(&file);
+    QString str = fs.readAll();
+
+    qDebug() << "------- IO ---------";
+    QTextStream stream(&str);
+    while (!stream.atEnd()) {
+        QString str = stream.readLine();
+
+        QString rchar("rchar: ");
+        if (str.startsWith(rchar)) {
+            ulong amt = str.mid(rchar.size()).toULong();
+            qDebug() << "Read:" << amt / 1024  << "kb";
+        }
+
+        QString wchar("wchar: ");
+        if (str.startsWith(wchar)) {
+            ulong amt = str.mid(wchar.size()).toULong();
+            qDebug() << "Write:" << amt / 1024  << "kb";
+        }
+
+        QString read("read_bytes: ");
+        if (str.startsWith(read)) {
+            ulong amt = str.mid(read.size()).toULong();
+            qDebug() << "Actual Reads:" << amt / 1024  << "kb";
+        }
+
+        QString write("write_bytes: ");
+        if (str.startsWith(write)) {
+            ulong amt = str.mid(write.size()).toULong();
+            qDebug() << "Actual Writes:" << amt / 1024  << "kb";
+        }
+    }
+    qDebug() << "\nThe actual read/writes may be 0 because of an existing"
+             << "cache and /tmp being memory mapped";
     quit();
 }
 
-- 
1.9.1

