From: Maximiliano Curia <maxy@gnuservers.com.ar>
Date: Wed, 7 Dec 2016 14:30:43 +0100
Subject: Disable the file indexer by default

---
 src/file/fileindexerconfig.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/file/fileindexerconfig.cpp b/src/file/fileindexerconfig.cpp
index 0e4f4c4..eae8ef5 100644
--- a/src/file/fileindexerconfig.cpp
+++ b/src/file/fileindexerconfig.cpp
@@ -350,7 +350,7 @@ void FileIndexerConfig::setDatabaseVersion(int version)
 
 bool FileIndexerConfig::indexingEnabled() const
 {
-    return m_config.group("Basic Settings").readEntry("Indexing-Enabled", true);
+    return m_config.group("Basic Settings").readEntry("Indexing-Enabled", false);
 }
 
 uint FileIndexerConfig::maxUncomittedFiles()
